IO/Ex1.cpp

```diff
@@ -0,0 +1,93 @@
+
+#include <stdexcept> //runtime_error
+#include <cassert>
+
+#include "Ex1.hpp"
+
+/************************************* Main Functions *******************************************/
+
+//Ctor
+FileHandler::FileHandler(const std::string path, std::ios_base::openmode permissions)
+: m_path(path), 
+  m_permissions(permissions)
+{
+
+    m_fd.open(path, permissions);
+    if (!m_fd.is_open()) {
+        throw std::runtime_error("Failed to open file: " + path);
+    }
+}
+
+//CCtor
+FileHandler::FileHandler(const FileHandler &other) 
+: m_path(other.m_path), 
+  m_permissions(other.m_permissions) {
+
+    m_fd.open(m_path, m_permissions);
+    if (!m_fd.is_open()) {
+        throw std::runtime_error("Failed to open file: " + m_path);
+    }   
+}
+
+//MCtor
+FileHandler::FileHandler(FileHandler &&other) noexcept 
+: m_fd(std::move(other.m_fd)),
+  m_path(std::move(other.m_path)),
+  m_permissions(other.m_permissions)
+{
+    other.m_permissions = std::ios_base::openmode(0);
+}
+
+//Copy Assignment
+FileHandler& FileHandler::operator=(const FileHandler &other) {
+
+    assert(this != &other);
```

Comment:
If you use `assert()` you need to make sure `NDEBUG` is defined in your testing build